sdk/core/Azure.Core/src/Pipeline/HttpPipelineMethod.cs

```diff
@@ -1,15 +1,100 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+
 namespace Azure.Core.Pipeline
 {
-    public enum HttpPipelineMethod
+    public readonly struct RequestMethod : IEquatable<RequestMethod>
     {
-        Get,
-        Post,
-        Put,
-        Patch,
-        Delete,
-        Head
+        public string Method { get; }
+
+        public static RequestMethod Get { get; } = new RequestMethod("GET");
+        public static RequestMethod Post { get; } = new RequestMethod("POST");
+        public static RequestMethod Put { get; } = new RequestMethod("PUT");
+        public static RequestMethod Patch { get; } = new RequestMethod("PATCH");
+        public static RequestMethod Delete { get; } = new RequestMethod("DELETE");
+        public static RequestMethod Head { get; } = new RequestMethod("HEAD");
+
+        public RequestMethod(string method)
```

Comment:
For the generated string based enum-like types, we also add conversions to/from string.  You can see an example at https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs#L8942. 

Not sure if it matters much here, but it wouldn't be bad to have a consistent pattern for these things across the SDK.

Reply:
I'm, honestly, not sure about this. I would like to avoid people passing strings instead of pre-defined request methods, it's both less efficient and rarely required.

Reply:
That's fair.